File: native/mixdog-graph/tests/fixtures/parity/php/sample.php
<?php
// Parity fixture: declaration and import shapes the graph reports for PHP.
namespace App\Sample;

use App\Models\User;
use App\Models\{Post, Comment};
use App\Contracts\Storage as StorageContract;

require_once __DIR__ . '/bootstrap.php';
require 'helpers.php';

interface Storage
{
    public function read(string $key): string;
}

trait Loggable
{
    public function log(string $message): void
    {
        error_log($message);
    }
}

enum Mode: string
{
    case Fast = 'fast';
    case Slow = 'slow';

    public function label(): string
    {
        return ucfirst($this->value);
    }
}

class Store implements Storage
{
    use Loggable;

    private string $name;

    public function __construct(string $name)
    {
        $this->name = $name;
    }

    public function read(string $key): string
    {
        return $this->name . ':' . $key;
    }

    private static function build(string $name): self
    {
        return new self($name);
    }
}

function top_level(User $user, StorageContract $storage): string
{
    return $storage->read((string) $user->id) . Post::class . Comment::class;
}

File: native/mixdog-graph/tests/fixtures/parity/rust/sample.rs
// Parity fixture: declaration and import shapes the graph reports for Rust.
use std::collections::HashMap;
use std::io::{Read, Write};

mod helpers;

pub const LIMIT: usize = 8;
static REGISTRY: Option<&str> = None;

pub type Pairs = HashMap<String, usize>;

pub struct Store {
    pub name: String,
    size: usize,
}

pub enum Mode {
    Fast,
    Slow(u8),
}

pub trait Storage {
    fn read(&self, key: &str) -> Option<String>;

    fn write(&mut self, key: &str) -> bool {
        let _ = key;
        false
    }
}

impl Store {
    pub fn new(name: String) -> Self {
        fn inner_helper(value: usize) -> usize {
            value + 1
        }
        Self {
            name,
            size: inner_helper(0),
        }
    }

    fn size(&self) -> usize {
        self.size
    }
}

impl Storage for Store {
    fn read(&self, key: &str) -> Option<String> {
        let _ = (key, REGISTRY, LIMIT);
        None
    }
}

pub fn run<R: Read, W: Write>(input: &mut R, output: &mut W) -> std::io::Result<()> {
    let mut buffer = String::new();
    input.read_to_string(&mut buffer)?;
    output.write_all(buffer.as_bytes())
}

macro_rules! shout {
    ($value:expr) => {
        format!("{}!", $value)
    };
}

#[cfg(test)]
mod tests {
    use super::*;

    fn fixture() -> Store {
        Store::new("fixture".to_string())
    }

    #[test]
    fn reads_nothing() {
        assert!(fixture().read("k").is_none());
        assert_eq!(shout!("a"), "a!");
        assert_eq!(Mode::Fast as u8, 0);
    }
}
